src/optimagic/optimization/create_optimization_problem.py

```diff
@@ -474,8 +481,9 @@ def create_optimization_problem(
         if not isinstance(bounds, Bounds | None):
             raise ValueError("bounds must be a Bounds object or None")
 
-        if not isinstance(constraints, list | dict):
-            raise ValueError("constraints must be a list or a dictionary")
+        if not all(isinstance(c, dict) for c in constraints):
+            # TODO: Only allow list[Constraint]
+            raise ValueError("constraints must be a list of dictionaries")
```

Comment:
This is confusing now that dict are deprecated. Maybe write: "constraints must be a list of **Constraint objects or** dictionaries.")